packages/wrangler/src/d1/execute.tsx

```diff
@@ -200,10 +200,10 @@ async function executeLocally(
 			logger.log
 		);
 
-	if (!existsSync(dbDir) && shouldPrompt) {
-		const ok = await confirm(
-			`About to create ${readableRelative(dbPath)}, ok?`
-		);
+	if (!existsSync(dbDir)) {
+		const ok =
+			!shouldPrompt ||
+			(await confirm(`About to create ${readableRelative(dbPath)}, ok?`));
```

Comment:
Could we please get a test that covers the bug, for posterity 😄

Reply:
yeah I considered that and asked @rozenmd about it, if I understood correctly he did mention that there isn't a proper test suite for d1 commands and that we could go with the fix as is.

If you want I could look into adding a test now but I'd have to  set up the test suite (to some degree at least) I guess but since I am not familiar with things in this codebase nor I am really involved in d1 I'd imagine that it would be better for someone else to start (and I can add the test at that point?).

or I can give it a shot now if you think that'd be ok, I would be up for it :slightly_smiling_face:

Reply:
> yeah I considered that and asked @rozenmd about it, if I understood correctly he did mention that there isn't a proper test suite for d1 commands and that we could go with the fix as is.
> 
> If you want I could look into adding a test now but I'd have to set up the test suite (to some degree at least) I guess but since I am not familiar with things in this codebase nor I am really involved in d1 I'd imagine that it would be better for someone else to start (and I can add the test at that point?).
> 
> or I can give it a shot now if you think that'd be ok, I would be up for it 🙂

If you are willing to try, I am willing to help as well. Anything we both can't figure out we can kick towards Max 😆

Reply:
Additionally if you want to just get this fix in and do a fast follow Test (since none previously existed) I would support that in the particular case given the circumstances.

Reply:
ah sorry, I just now say the comment about merging the PR anyways, thanks :slightly_smiling_face::+1: